app/src/main/java/com/example/lastfm/ArtistListFragment.java

```diff
@@ -0,0 +1,103 @@
+package com.example.lastfm;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.SingleObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class ArtistListFragment extends Fragment {
+    private final int NUMBER_OF_ARTISTS = 40;
+    private List<Artist> requestedArtists = new ArrayList<>();
+    private SwipeRefreshLayout mSwipeRefreshLayout;
+
+    private ArtistAdapter artistsAdapter;
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        View view = inflater.inflate(R.layout.artist_list_fragment,container,false);
+        initRecyclerView(view);
+        retrofitRequest();
+        mSwipeRefreshLayout = view.findViewById(R.id.swiperefresh);
+        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                retrofitRequest();
+            }
+        });
+        return view;
+    }
+
+    private void retrofitRequest() {
+        LastFMClient client = ServiceGenerator.getInstance().getLastFMClient();
+        client.getArtists(NUMBER_OF_ARTISTS)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new SingleObserver<List<Artist>>() {
+                        @Override
+                        public void onSubscribe(Disposable d) {
+                        }
+
+                        @Override
+                        public void onSuccess(List<Artist> info) {
+                            requestedArtists = info;
+                            showArtists(requestedArtists);
+                            hideRefreshing();
+                        }
+
+                        @Override
+                        public void onError(Throwable e) {
+                            e.printStackTrace();
+                            hideRefreshing();
+                            String requestErrorText = getString(R.string.request_error_message);
```

Comment:
Why do u not show this message to a user?

Reply:
For knowing that loading of artist was ended with failure (no internet connection & etc.)